salt/modules/cohesity.py

```diff
@@ -0,0 +1,554 @@
+# c_username = cohesity_config.get("username", "")
+# c_password = cohesity_config.get("password", "")
+# c_domain = cohesity_config.get("domain", "")
+# cohesity_client = CohesityClient(
+#     cluster_vip=cluster_vip, username=c_username, password=c_password, domain=c_domain,
+# )
+
+
+def __virtual__():
+    if HAS_LIBS:
+        return __virtualname__
+    return True
+
+
+def get_sd_id(name):
+    """
+    Function to fetch storage domain available in the cluster.
+    : return storage domain id.
+    """
+    cohesity_client = _get_client()
+    log.info("Getting sorage domain with name {}".format(name))
+    resp = cohesity_client.view_boxes.get_view_boxes(names=name)
+    if resp:
+        return resp[0].id
+
+
+def get_policy_id(name):
+    """
+    Function to fetch policy available in the cluster.
+    : return policy id.
+    """
+    cohesity_client = _get_client()
+    log.info("Getting policy with name {}".format(name))
+    resp = cohesity_client.protection_policies.get_protection_policies(names=name)
+    if resp:
+        return resp[0].id
+
+
+def get_vmware_source_ids(name, vm_list):
+    """
+    Function to virtual machines available in the vcenter.
+    : return source ids and vcenter id.
+    """
+    cohesity_client = _get_client()
+    source_id_list = []
+    parent_id = -1
+    log.info("Fetching Vcenter and Vm ids")
+    try:
+        result = cohesity_client.protection_sources.list_protection_sources_root_nodes(
+            environments=env_enum.K_VMWARE
+        )
+        for each_source in result:
+            endpoint = each_source.registration_info.access_info.endpoint
+            v_name = each_source.protection_source.name
+
+            # Check for both endpoint and source name.
+            if name in [endpoint, v_name]:
+                parent_id = each_source.protection_source.id
+        if parent_id == -1:
+            log.error("Vcenter {} not available in the cluster".format(name))
+            exit()
+        vms = cohesity_client.protection_sources.list_virtual_machines(
+            v_center_id=parent_id, names=vm_list
+        )
+        vm_names = copy.deepcopy(vm_list)
+        for vm in vms:
+            vm_names.remove(vm.name)
+            source_id_list.append(vm.id)
+        if vm_names:
+            log.error(
+                "Following list of vms '{}' are not available in vcenter, "
+                "please make sure the virtual machine names are correct".format(
+                    ",".join(vm_names)
+                )
+            )
+        return parent_id, source_id_list
+    except APIException as err:
+        log.error(err)
+        return -1, []
+
+
+def register_vcenter(vcenter, username, password):
+    """
+    Function to fetch register Vmware Vcenter to cohesity cluster.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+    salt-call cohesity.register_vcenter vcenter=vcenter_name username=admin password=admin
+    """
```

Comment:
```suggestion
def register_vcenter(vcenter, username, password):
    """
    Function to fetch register Vmware Vcenter to cohesity cluster.

    CLI Examples:

    .. code-block:: bash

        salt-call cohesity.register_vcenter vcenter=vcenter_name username=admin password=admin
    """
```

Reply:
Apply the same ident on all your other `.. code-block:: bash`

Reply:
Done